Sources/NIOSSH/Child Channels/SSHChildChannel.swift

```diff
@@ -1184,3 +1188,16 @@ private extension IOData {
         }
     }
 }
+
+extension SSHChildChannel {
+    /// This is a ``Sendable`` box around a ``ChannelPipeline``. This is needed since the
+    /// ``SSHChildChannel`` needs to have an IUO to the pipeline because the pipeline needs a
```

Comment:
The `SSHChildChannel` is not marked as `Sendable`; should it be?

Reply:
It's not clear to me why this is necessary: I think we should prefer `@unchecked Sendable` until such time as `ChannelPipeline` is made `Sendable`.

Reply:
Even if `ChannelPipeline` is `Sendable`, `SSHChildChannel` would need to be `@unchecked Sendable` (because `var _pipeline: ChannelPipeline!` is mutable). 

I think the idea here was to explicitly mark this wrapper around `ChannelPipeline` as `@unchecked Sendable` so that everything else in `SSHChildChannel` has `Sendable` checks. Maybe that's overkill though 🤷‍♂️.

Reply:
Hmm, I _suspect_ it's overkill, but I could be talked out of it.

Reply:
I see why it might be overkill but isn't it better to extract this single property out so that in the future if somebody adds something to the `SSHChildChannel` which would break sendability he will get a warning. Whereas if we make the whole channel `@unchecked Sendable` all future warnings will be silenced .

Reply:
I don't think @glbrntt's argument holds water, because the same argument applies to `PipelineSendableBox`: it has not magically made the mutability of the IUO go away. This appears to me to be "tricking" the compiler into thinking the property has suddenly become thread-safe, but the thread safety of this property has nothing to do with the box and everything to do with the fact that we don't publish a `self` reference anywhere except the `ChannelPipeline` before the `SSHChildChannel` is fully initialised: that is, we never write this field. I think the fact that this defeats `Sendable` checking is actually a compiler bug, and we should assume it will be fixed. This type does need to be `@unchecked Sendable`, because the compiler will struggle to prove that we're not doing anything dodgy with the IUO.